### PR TITLE
Refactor some uses of the blockless `String#split`

### PR DESCRIPTION
**Pull request by HertzDevil**
The ones followed by `Array#each` can use the block overload directly. Some others can be replaced by `String#partition`.

### CODE REVIEW

File: src/http/formdata.cr [hunk 1]
```diff
@@ -138,13 +138,15 @@ module HTTP::FormData
...
-    raise Error.new("Invalid Content-Disposition: not form-data") unless type == "form-data"
-    (1...parts.size).each do |i|
-      part = parts[i]
+    first = true
+    content_disposition.split(';') do |part|
+      if first
+        first = false
+        raise Error.new("Invalid Content-Disposition: not form-data") unless part == "form-data"
+        next
+      end
```
Comment (straight-shoota) [inline src/http/formdata.cr:147]: **suggestion:**
This is much cleaner and performance characteristics should be comparable:
```suggestion
    parts = content_disposition.lchop?("form-data;")
    raise Error.new("Invalid Content-Disposition: not form-data") unless parts
    parts.split(';') do |part|
```

**Review by straight-shoota — COMMENTED** [review]
I've wanted to do this for a long time. Thanks for picking it up!

File: src/http/formdata.cr (hunk 1 above)
Comment (HertzDevil) [reply src/http/formdata.cr:147]: That would create a near copy of `content_disposition` which may consume even more memory than the original implementation.

Maybe the existing use of `.split` isn't so bad after all?

Comment (straight-shoota) [reply src/http/formdata.cr:147]: Good point. Maybe we could avoid that allocation by using `starts_with?` and nothing else. Unknown keys are implicitly skipped in the loop.
However, there is the issue of potential white space around the separators (not sure if that's valid though).
So yeah probably we should better leave this unchanged for now. It's growing too much complexity for a simple generic refactor.

File: src/http/formdata.cr [hunk 2]
```diff
@@ -138,15 +138,11 @@ module HTTP::FormData
...
-        raise Error.new("Invalid Content-Disposition: not form-data") unless part == "form-data"
-        next
-      end
-
-      key, _, value = part.partition('=')
+    parts = content_disposition.split(';')
+    type = parts.shift?
+    raise Error.new("Invalid Content-Disposition: not form-data") unless type == "form-data"
+    parts.each do |part|
+      key, value = part.split('=', 2)
```
Comment (straight-shoota) [inline src/http/formdata.cr:145]: I think `partition` could stay here?

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**